src/common/components/navigation/MenuItem.tsx

```diff
@@ -1,44 +1,74 @@
-import {
-  ListItemIcon,
-  Tooltip,
-  Theme,
-  createStyles,
-  makeStyles,
-} from "@material-ui/core";
+import { ListItemIcon, Tooltip, makeStyles, Theme } from "@material-ui/core";
 import ListItem from "@material-ui/core/ListItem";
 import ListItemText from "@material-ui/core/ListItemText";
 import React, { ComponentClass, ElementType, ReactElement } from "react";
-import { NavLink, useLocation, useRouteMatch } from "react-router-dom";
+import { NavLink, useRouteMatch } from "react-router-dom";
 import { Path } from "../../../router/Path";
 
 export type MenuItemProps = {
   path: Path;
   text: string;
-  component: ComponentClass<any> | (() => ReactElement);
+  component:
+    | ComponentClass<any>
+    | (() => ReactElement)
+    | React.ReactElement<any>;
   icon: ElementType;
   isFallback?: boolean;
   isDisabled?: boolean;
   tooltipTextRows?: string[];
+  isBeforeSelected?: boolean;
+  isAfterSelected?: boolean;
+  selected?: boolean;
 };
 
-const useStyles = makeStyles((theme: Theme) =>
-  createStyles({
-    disabled: {
-      color: theme.palette.text.disabled,
+const getListItemColor = (theme: Theme, props: MenuItemProps) => {
+  return props.isDisabled
+    ? theme.palette.text.disabled
+    : props.selected
+    ? "#d7d9e2"
+    : "#9a9ca4";
```

Comment:
Would prefer if-else for sake of readability when there are 2+ ternary expressions nested.